src/code/fv-components/egg-hatch-drake.js

```diff
@@ -0,0 +1,53 @@
+import React, { PropTypes } from 'react';
+import AnimatedSprite from '../components/animated-sprite';
+import OrganismView from '../components/organism';
+import classNames from 'classnames';
+
+export default class EggHatchDrakeView extends React.Component {
+
+  static propTypes = {
+    id: PropTypes.string,
+    eggClasses: PropTypes.string,
+    drakeClasses: PropTypes.string,
+    drake: PropTypes.object,
+    hatchStarted: PropTypes.bool,
+    hatchComplete: PropTypes.bool,
+    onHatchComplete: PropTypes.func
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
```

Comment:
That makes sense about simplifying the client code, I can see using one component per challenge in that case.

I think I realized what's really making me uncomfortable about the code is that `hatchComplete` actually means multiple things. The `hatchComplete` prop is really used to communicate whether or not to skip the animation, and the `hatchComplete` state is used to track if the animation is finished. I would advocate for changing the name of the prop to something like `skipAnimation`, and leaving the rest of the code the way it is.

This clarifies the use of each variable in my mind, and lines them up better with the way they're used. What do you think?